packages/nodes-base/nodes/Schedule/test/GenericFunctions.test.ts

```diff
@@ -0,0 +1,132 @@
+import * as n8nWorkflow from 'n8n-workflow';
+import { recurrenceCheck, toCronExpression } from '../GenericFunctions';
+import type { IRecurrenceRule } from '../SchedulerInterface';
+
+describe('toCronExpression', () => {
+	Object.defineProperty(n8nWorkflow, 'randomInt', {
+		value: (min: number, max: number) => Math.floor((min + max) / 2),
+	});
+
+	it('should return cron expression for cronExpression field', () => {
+		const result = toCronExpression({
+			field: 'cronExpression',
+			expression: '1 2 3 * * *',
+		});
+		expect(result).toEqual('1 2 3 * * *');
+	});
+
+	it('should return cron expression for seconds interval', () => {
+		const result = toCronExpression({
+			field: 'seconds',
+			secondsInterval: 10,
+		});
+		expect(result).toEqual('*/10 * * * * *');
+	});
+
+	it('should return cron expression for minutes interval', () => {
+		const result = toCronExpression({
+			field: 'minutes',
+			minutesInterval: 30,
+		});
+		expect(result).toEqual('30 */30 * * * *');
+	});
+
+	it('should return cron expression for hours interval', () => {
+		const result = toCronExpression({
+			field: 'hours',
+			hoursInterval: 3,
+			triggerAtMinute: 22,
+		});
+		expect(result).toEqual('30 22 */3 * * *');
+
+		const result1 = toCronExpression({
+			field: 'hours',
+			hoursInterval: 3,
+		});
+		expect(result1).toEqual('30 30 */3 * * *');
+	});
+
+	it('should return cron expression for days interval', () => {
+		const result = toCronExpression({
+			field: 'days',
+			daysInterval: 4,
+			triggerAtMinute: 30,
+			triggerAtHour: 10,
+		});
+		expect(result).toEqual('30 30 10 * * *');
+
+		const result1 = toCronExpression({
+			field: 'days',
+			daysInterval: 4,
+		});
+		expect(result1).toEqual('30 30 12 * * *');
+	});
+
+	it('should return cron expression for weeks interval', () => {
+		const result = toCronExpression({
+			field: 'weeks',
+			weeksInterval: 2,
+			triggerAtMinute: 0,
+			triggerAtHour: 9,
+			triggerAtDay: [1, 3, 5],
+		});
+		expect(result).toEqual('30 0 9 * * 1,3,5');
+		const result1 = toCronExpression({
+			field: 'weeks',
+			weeksInterval: 2,
+			triggerAtDay: [1, 3, 5],
+		});
+		expect(result1).toEqual('30 30 12 * * 1,3,5');
+	});
+
+	it('should return cron expression for months interval', () => {
+		const result = toCronExpression({
+			field: 'months',
+			monthsInterval: 3,
+			triggerAtMinute: 0,
+			triggerAtHour: 0,
+			triggerAtDayOfMonth: 1,
+		});
+		expect(result).toEqual('30 0 0 1 */3 *');
+		const result1 = toCronExpression({
+			field: 'months',
+			monthsInterval: 3,
+		});
+		expect(result1).toEqual('30 30 12 15 */3 *');
+	});
+});
+
+describe.only('recurrenceCheck', () => {
```

Comment:
@ivov any idea why this did not get caught by the linter?

Reply:
Likely we need to update the first path in [this line](https://github.com/n8n-io/n8n/blob/3d0393c739bd6727446f946e7ecbf79fe7cfd5bf/packages/@n8n_io/eslint-config/base.js#L470) to `'**/test/**/*.ts'`. Or have nodes-base start using the pattern from other packages: `**/__tests__/*.ts`